### PR TITLE
[BUG-7732][fix] fix column 'is_directory' of table t_ds_resources typ…

### LINKED ISSUE #7732: [Bug] [SQL] field ‘is_directory’ in t_ds_resources table has error type in PostgreSQL database
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image

create directory error logs:
```
### Error updating database.  Cause: org.postgresql.util.PSQLException: ERROR: column "is_directory" is of type integer but expression is of type boolean
  Hint: You will need to rewrite or cast the expression.
  Position: 192
### The error may exist in org/apache/dolphinscheduler/dao/mapper/ResourceMapper.java (best guess)
### The error may involve org.apache.dolphinscheduler.dao.mapper.ResourceMapper.insert-Inline
### The error occurred while setting parameters
### SQL: INSERT INTO t_ds_resources  ( file_name, size, create_time, description, full_name, alias, update_time, pid, type, user_id, is_directory )  VALUES  
( ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? )
### Cause: org.postgresql.util.PSQLException: ERROR: column "is_directory" is of type integer but expression is of type boolean
  Hint: You will need to rewrite or cast the expression.
  Position: 192
; bad SQL grammar []; nested exception is org.postgresql.util.PSQLException: ERROR: column "is_directory" is of type integer but expression is of type boolean
```

### What you expected to happen

need change the field ‘is_directory’ type from int to bool,then ok

### How to reproduce

when create directory

### Anything else

_No response_

### Version

2.0.1

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jim-parsons**
## Purpose of the pull request

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add `Boolean2IntTypeHandler.java` both in class `Resource.java` and `ResourceMapper.xml`
2. specify typeHandler packages and update session config in class `SpringConnectionFactory` and remove the deprecated bean `PaginationInterceptor`
3. format class `Resource`
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). 
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type's null situation.
6. update `known-dependencies.txt`

This closes #7732

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@
         <hadoop.version>2.7.3</hadoop.version>
         <quartz.version>2.3.2</quartz.version>
         <jackson.version>2.10.5</jackson.version>
-        <mybatis-plus.version>3.2.0</mybatis-plus.version>
+        <mybatis-plus.version>3.4.0</mybatis-plus.version>
```
Comment (ruanwenjun) [inline pom.xml:63]: It's better not upgrade `mybatis-plus` in this PR, If you hope to do this, it's necessary to create a new issue and discuss with community first.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (jim-parsons) [reply pom.xml:63]: sorry about that, i choose the way of adding a typeHandler to fix issue #7732 . but due to the mybatis-plus [BUG](https://github.com/baomidou/mybatis-plus/issues/2550), it failed in the maven test. so i update a oldest version that works.
  
  but if there is anthor way to fix this bug, i think i won't change the version

Comment (ruanwenjun) [reply pom.xml:63]: We can discuss in #7732 .

**Comment by jim-parsons** [comment]
this pr can close, thx, i've create a new one.

**Pull request closed without merging**